Snowflake.Data/Core/FileTransfer/SFFileTransferAgent.cs

```diff
@@ -667,155 +667,152 @@ private void initEncryptionMaterial()
-                        * For example, "*.xls" returns both "book.xls" and "book.xlsx".
-                        * - In all other cases, the method returns files that exactly match the 
-                        * specified extension. For example, "*.ai" returns "file.ai" but not "file.aif".
+                        /*
+                         * When you use the asterisk wildcard character in a searchPattern such as
+                         * "*.txt", the method returns files that exactly match the specified extension.
+                         * For example, "*.ai" returns "file.ai" but not "file.aif".
                         */
-                    string[] potentialMatches =
+                        var potentialMatches =
                             Directory.GetFiles(
-                                directoryName,
+                                directory,
                                 fileName,
                                 SearchOption.TopDirectoryOnly);
-                    foreach (string potentialMatch in potentialMatches)
+                        filePaths.AddRange(potentialMatches.Where(potentialMatch => potentialMatch.EndsWith(ext)));
+                    }
+                    else
                     {
-                        if (potentialMatch.EndsWith(ext))
-                        {
-                            filePaths.Add(potentialMatch);
-                        }
+                        // If there is a wildcard in the file name in the file path
+                        filePaths.AddRange(
+                            Directory.GetFiles(
+                                directory, 
+                                fileName, 
+                                SearchOption.TopDirectoryOnly));
                     }
                 }
-                else
-                {
-                    // If there is a wildcard in the file name in the file path
-                    filePaths.AddRange(
-                        Directory.GetFiles(
-                            directoryName,
-                            fileName,
-                            SearchOption.TopDirectoryOnly));
-                }
             }
             else
             {
-                // No wild card, just make sure it's a file
-                FileAttributes attr = File.GetAttributes(location);
-                if (!attr.HasFlag(FileAttributes.Directory))
+                // No wildcard in the filename
+                foreach (var directory in foundDirectories)
                 {
-                    filePaths.Add(location);
-                }
-                else
-                {
-                    throw new FileNotFoundException(
-                        "Directories not supported, you need to provide a file path", location);
+                    var fullPath = Path.GetFullPath(directory + fileName);
+                    // Make sure it's a file
```

Comment:
let's create a method for that

Reply:
Created a method `IsDirectory(string path)`.